scripts/kvm-test.py

```diff
@@ -174,44 +170,31 @@ def load_config(self):
                     help='boot test image')
 
 
-def waitstatus_to_exitcode(waitstatus):
-    '''If the process exited normally (if WIFEXITED(status) is true), return
-    the process exit status (return WEXITSTATUS(status)): result greater
-    than or equal to 0.
-
-    If the process was terminated by a signal (if WIFSIGNALED(status) is
-    true), return -signum where signum is the number of the signal that
-    caused the process to terminate (return -WTERMSIG(status)): result less
-    than 0.
-
-    Otherwise, raise a ValueError.'''
-
-    # This function is for python 3.9 compat
-
-    if 'waitstatus_to_exitcode' in dir(os):
-        return os.waitstatus_to_exitcode(waitstatus)
-    if os.WIFEXITED(waitstatus):
-        return os.WEXITSTATUS(waitstatus)
-    if os.WIFSIGNALED(waitstatus):
-        return -os.WTERMSIG(waitstatus)
-
-    raise ValueError
-
-
-class SubProcessFailure(Exception):
-    pass
+def parse_args():
+    ctx = Context(parser.parse_args())
+    if ctx.args.quick or ctx.args.basesnap or ctx.args.snap \
+            or ctx.args.channel or ctx.args.reuse:
+        ctx.args.build = True
+    if ctx.args.reuse:
+        ctx.args.save = True
+    return ctx
+
+
+def run(cmd):
+    if isinstance(cmd, str):
+        cmd_str = cmd
+        cmd_array = cmd.split(' ')
```

Comment:
In a non-serious script like this I prefer the readability of the string, so I'm leveraging the shlex calls.

Reply:
Oh, what I meant in my second comment:

I didn't realize that the `print(f'+ {cmd}')` instruction was not something new in this PR. Github showed it as a new line because the context changed.